pkg/trace/agent/agent.go

```diff
@@ -496,13 +501,8 @@ func (a *Agent) runSamplers(now time.Time, pt traceutil.ProcessedTrace, hasPrior
 // ErrorSampler are run in parallel. The RareSampler catches traces with rare top-level
 // or measured spans that are not caught by PrioritySampler and ErrorSampler.
 func (a *Agent) samplePriorityTrace(now time.Time, pt traceutil.ProcessedTrace) bool {
-	var rare bool
-	if !a.conf.RareSamplerEnabled {
-		rare = false
-	} else {
-		// run this early to make sure the signature gets counted by the RareSampler.
-		rare = a.RareSampler.Sample(now, pt.TraceChunk, pt.TracerEnv)
-	}
+	// run this early to make sure the signature gets counted by the RareSampler.
+	rare := a.RareSampler.Sample(now, pt.TraceChunk, pt.TracerEnv)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ Question:** ‏Just for my personal understanding here : we enable the `RareSampler` by default now?

Reply:
yes we do and that's new, good catch :)

Reply:
It's not done here, it was done by APM agent's team :)